bin/jenkins-dist-publish.sh

```diff
@@ -5,6 +5,37 @@
 
 set -e -x -o pipefail
 
+# $1: profile (e.g. "hadoop-2.6")
+function does_profile_exist() {
+    (cd "${SPARK_DIR}" && ./build/mvn help:all-profiles | grep "$1")
+}
+
+# uploads build/spark/spark-*.tgz to S3
+function upload_to_s3 {
+    aws s3 cp --acl public-read "${SPARK_DIST_DIR}/${SPARK_DIST}" "${S3_URL}"
+}
+
+function set_hadoop_versions {
+    HADOOP_VERSIONS=( "2.4" "2.6" "2.7" )
+}
+
+# rename build/dist/spark-*.tgz to build/dist/spark-<TAG>.tgz
+# globals: $SPARK_VERSION
+function rename_dist {
+    SPARK_DIST_DIR="spark-${SPARK_VERSION}-bin-${HADOOP_VERSION}"
+    SPARK_DIST="${SPARK_DIST_DIR}.tgz"
+
+    mkdir -p "${SPARK_DIST_DIR}"
+
+    pushd "${DIST_DIR}"
+    tar xvf spark-*.tgz
+    rm spark-*.tgz
```

Comment:
Can one use `tar --transform` here (https://www.gnu.org/software/tar/manual/html_section/tar_52.html) instead of extracting and then recreating the archive?